pkg/appgw/configbuilder.go

```diff
@@ -95,21 +98,36 @@ func generateListenerID(rule *v1beta1.IngressRule,
 	return listenerID
 }
 
-// Build generates the ApplicationGatewayPropertiesFormat for azure resource manager
-func (c *appGwConfigBuilder) Build(ingressList []*v1beta1.Ingress, serviceList []*v1.Service) (*network.ApplicationGatewayPropertiesFormat, error) {
-	return &c.appGwConfig, c.Validate(ingressList, serviceList)
+// GetApplicationGatewayPropertiesFormatPtr gets a pointer to updated ApplicationGatewayPropertiesFormat.
+func (c *appGwConfigBuilder) GetApplicationGatewayPropertiesFormatPtr() *network.ApplicationGatewayPropertiesFormat {
+	return &c.appGwConfig
 }
 
-// Validate runs all the validators on the config constructed to ensure it complies with App Gateway requirements.
-func (c *appGwConfigBuilder) Validate(ingressList []*v1beta1.Ingress, serviceList []*v1.Service) error {
+// PreBuildValidate runs all the validators that suggest misconfiguration in Kubernetes resources.
+func (c *appGwConfigBuilder) PreBuildValidate(envVariables environment.EnvVariables, ingressList []*v1beta1.Ingress, serviceList []*v1.Service) error {
 
-	validators := []func(eventRecorder record.EventRecorder, config *network.ApplicationGatewayPropertiesFormat, ingressList []*v1beta1.Ingress, serviceList []*v1.Service) error{
-		validateURLPathMaps,
+	validators := []func(eventRecorder record.EventRecorder, config *network.ApplicationGatewayPropertiesFormat, envVariables environment.EnvVariables, ingressList []*v1beta1.Ingress, serviceList []*v1.Service) error{
```

Comment:
The bodies of the two validation groups seem awfully similar - there might be some opportunity to DRY this ;)

Reply:
Good point!